### PR TITLE
Encode bignums as bytes

### PR DESCRIPTION
**Pull request by j16r**

### CODE REVIEW

File: contracts/CBOR.sol [hunk 1]
```diff
@@ -52,6 +60,56 @@ library CBOR {
...
+      encodeBytes(buf, encodeUnsignedBigNum(uint(value)));
+    }
+
+    function encodeSignedBigNum(Buffer.buffer memory buf, int input) internal pure {
+      buf.appendUint8(uint8((MAJOR_TYPE_TAG << 5) | TAG_TYPE_NEGATIVE_BIGNUM));
+      encodeBytes(buf, encodeUnsignedBigNum(uint(-1 - input)));
+    }
+
+    function encodeUnsignedBigNum(uint value) internal pure returns (bytes) {
+      uint8 size = byteCount(value);
```
Comment (se3000) [inline contracts/CBOR.sol]: Would it be valid CBOR to assume a max size and always use that, padded with zeros? I imagine we'd save on computation, but pay a small amount more on memory.

File: contracts/CBOR.sol [hunk 2]
```diff
@@ -40,7 +44,11 @@ library CBOR {
     }
 
     function encodeInt(Buffer.buffer memory buf, int value) internal pure {
-        if(value >= 0) {
+        if(value < -10000000000000000) {
```
Comment (samsondav) [inline contracts/CBOR.sol]: Interesting, why this magic number?

File: contracts/CBOR.sol (hunk 1 above)
Comment (j16r) [reply contracts/CBOR.sol]: I don't think so, maybe a little antithetical to the saving storage philosophy of CBOR, but not against the spec.

File: contracts/CBOR.sol (hunk 2 above)
Comment (j16r) [reply contracts/CBOR.sol]: Should be the boundary between 8 byte and > 8 byte integers.

Comment (j16r) [reply contracts/CBOR.sol]: I may add some boundary tests to make sure I have this right, I may have a fencepost bug here.

Comment (se3000) [reply contracts/CBOR.sol]: I think you might need to prepend a `0x`. Looks like a boundary on hex but not decimals.

Comment (j16r) [reply contracts/CBOR.sol]: Yep you're right, fixed.

**Review by alexroan — APPROVED** [review]

File: contracts/Migrations.sol [hunk 3]
```diff
@@ -1,4 +1,4 @@
-pragma solidity ^0.4.17;
+pragma solidity >= 0.4.19 < 0.7.0;
```
Comment (se3000) [inline contracts/Migrations.sol:1]: Did not know you could do a range like that.

**Review by se3000 — APPROVED** [review]

**Pull request merged**